src/include/platform/DeviceInstanceInfoProvider.h

```diff
@@ -194,6 +194,22 @@ class DeviceInstanceInfoProvider
     {
         return CHIP_ERROR_NOT_IMPLEMENTED;
     }
+
+    /**
+     * @brief Obtain a Software Version String from the device's factory data.
+     *
+     * The SoftwareVersionString can be used to provide a more user-friendly value than that
+     * represented by the SoftwareVersion attribute.
+     *
+     * @param[in, out] buf Buffer to copy string.
+     *                 On CHIP_NO_ERROR return from this function this buffer will be null-terminated.
+     *                 On error CHIP_ERROR_BUFFER_TOO_SMALL there is no guarantee that buffer will be null-terminated.
+     * @param[in] bufSize Size of data, including the null terminator, that can be written to buf.
+     *                    This size should be +1 higher than maximum possible string.
+     * @returns CHIP_NO_ERROR on success, or another CHIP_ERROR from the underlying implementation
+     *          if access fails.
+     */
+    virtual CHIP_ERROR GetSoftwareVersionString(char * buf, size_t bufSize) { return CHIP_ERROR_NOT_IMPLEMENTED; }
```

Comment:
What's the rationale for adding this method to `DeviceInstanceInfoProvider`? Software version is a property of the firmware, not of a device instance, so it's pretty unlikely that a manufacturer puts this value to the factory data as its needs to be updated every software update.
Moreover, now we have one interface to obtain Software Version and another one to obtain Software Version String, which doesn't seem intuitive :).
@feasel0 @tcarmelveilleux